pkg/sql/alter_sequence.go

```diff
@@ -150,6 +153,18 @@ func (n *alterSequenceNode) startExec(params runParams) error {
 			} else {
 				restartVal = &opts.Start
 			}
+		} else if option.Name == tree.SeqOptCacheNode {
+			if !params.p.execCfg.Settings.Version.IsActive(params.ctx, clusterversion.V24_1) {
+				return pgerror.New(
+					pgcode.FeatureNotSupported,
+					`node-level cache not supported before V24.1`,
```

Comment:
Could we add on to your mixed-version tests so that this code will get tested as well